### PR TITLE
Add forced update

### PR DESCRIPTION
**Pull request by borisyankov**
On startup do a check to an url loaded from config.js
If the response is anything but status === 400, it is ok (200 or 404 are ok).
If not compatible show a specialized screen.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@borisyankov, it looks like you've already signed the Dropbox CLA. Thanks!

File: src/nav/Navigation.js [hunk 1]
```diff
@@ -51,6 +52,8 @@ export default class Navigation extends React.Component {
 
   renderScene = (props) => {
     switch (props.scene.route.key) {
+      case 'compatibility':
+        return <CompatibilityScreen />;
```
Comment (neerajwahi) [inline src/nav/Navigation.js]: Does NavigationExperimental have a concept of modals?

File: src/start/CompatibilityScreen.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  }
+});
+
+export default class CompatibilityScreen extends React.Component {
+
+  render() {
+    return (
+      <View style={styles.screen}>
+        <Logo />
+        <Text style={styles.text}>This app is too old!</Text>
```
Comment (neerajwahi) [inline src/start/CompatibilityScreen.js]: We can change the message itself later. Maybe something more along the lines of 'There's a newer version of the app available. Please upgrade to continue' with a link.

File: src/nav/NavigationContainer.js [hunk 3]
```diff
@@ -2,28 +2,39 @@ import React from 'react';
 import { connect } from 'react-redux';
 
 import boundActions from '../boundActions';
+import { checkCompatibility } from '../api';
 import { getInitialRoutes } from './routingSelectors';
 import Navigation from './Navigation';
 
 class NavigationContainer extends React.PureComponent {
 
+  state = {
```
Comment (neerajwahi) [inline src/nav/NavigationContainer.js]: Could you please add a comment or two that `compatibilityCheckFail` has to do with forced upgrades? Someone new to the code might be confused about its purpose

File: src/nav/Navigation.js (hunk 1 above)
Comment (borisyankov) [inline src/nav/Navigation.js]: Yeah, though this line is no longer needed, will remove.

File: src/start/CompatibilityScreen.js (hunk 2 above)
Comment (borisyankov) [inline src/start/CompatibilityScreen.js]: I have added a task or later. When we publish both the iOS and Android apps. We don't have the links yet, so leaving as is for now.

Comment (borisyankov) [inline src/start/CompatibilityScreen.js]: That is not descriptive enough. We are firing this not on update available but on `mandatory` update.

**Pull request merged**